test/labels.js

```diff
@@ -0,0 +1,349 @@
+    tc.fakeNodes[0].labels = {
+        "hello": "world"
+    };
+    callback();
+}, prepareCluster(function(t, tc, n) { return [
+        dsl.assertStats(t, tc, n+1, 0, 0, {
+            0: {
+                labels: { "hello": "world" }
+            }
+        }),
+    ];})
+);
+
+test2('ringpop should accept labels for a node with a higher incarnation number from the node itself', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) { return [
+        // feed the sut with label information for nodeIx 0
+        dsl.changeStatus(t, tc, 0, 0, {
+            subjectIncNoDelta: +1,
+            status: 'alive',
+            labels: {
+                "hello": "world"
+            },
+        }),
+        dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+        // assert that nodeIx 0 has its labels set on the sut
+        dsl.assertStats(t, tc, n+1, 0, 0, {
+            0: {
+                labels: {
+                    "hello": "world"
+                }
+            }
+        }),
+    ];})
+);
+
+test2('ringpop should accept labels for a node with a higher incarnation number from another node', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) { return [
+        // feed the sut with label information for nodeIx 0
+        dsl.changeStatus(t, tc, 0, 1, {
+            subjectIncNoDelta: +1,
+            status: 'alive',
+            labels: {
+                "hello": "world"
+            },
+        }),
+        dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+        // assert that nodeIx 0 has its labels set on the sut
+        dsl.assertStats(t, tc, n+1, 0, 0, {
+            1: {
+                labels: {
+                    "hello": "world"
+                }
+            }
+        }),
+    ];})
+);
+
+test2('ringpop should not accept labels for a node with a lower incarnation number', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) { return [
+        // feed the sut with label information for nodeIx 0
+        dsl.changeStatus(t, tc, 0, 1, {
+            subjectIncNoDelta: -1,
+            status: 'alive',
+            labels: {
+                "hello": "world"
+            },
+        }),
+        dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+        // assert that nodeIx 0 has its labels set on the sut
+        dsl.assertStats(t, tc, n+1, 0, 0, {
+            1: {
+                labels: undefined
+            }
+        }),
+    ];})
+);
+
+// The most likely assert to fail will be the last assert in this test where the
+// winning labels are compaired to the labels that were chosen after the first
+// round of label gossipping
+test2('ringpop should deterministically pick labels on conflict', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) {
+        // keep track of the labels that won the first time around
+        var winningLabels = {};
+
+        // two pairs of labels
+        var labels1 = { "hello": "world" };
+        var labels2 = { "hello": "goodbey" };
```

Comment:
Is `goodbey` a typo, or did you mean, like, you're telling your significant other that he/she has been good?

![beyonce wink](https://s-media-cache-ak0.pinimg.com/236x/a0/bf/a0/a0bfa0d51ea4d770adee65017d5088c1.jpg)